### DISCUSSION TITLE
Call a function after a Dialog closes with a Tailwind animation

### DISCUSSION THREAD

**Discussion by jordanthornquest** [adobe/react-spectrum#7139] [Q&A] [answered]
I'm currently working with the [React Aria Tailwind Starter](https://react-spectrum.adobe.com/react-aria-tailwind-starter) and React Router. All the components in this collection use `tailwind-animate`, `tailwind-variants`, and `tailwindcss-react-aria-components`. 

I'm rendering the `Dialog` component when a user visits a route (for example, `/new-post`). When a user presses a "Submit" or "Cancel" button within the Dialog, I'd like to wait for the `Dialog`'s closing animation to complete before navigating to a new route, which is accomplished by calling a `navigate()` function. 

How would this be accomplished with React Aria?

**Comment by reidbarber** [comment, marked as answer]
I would use [setTimeout](https://developer.mozilla.org/en-US/docs/Web/API/Window/setTimeout) to execute the `navigate()` function after a specified amount of time (i.e. 200ms for the tailwind starters) 

```ts
setTimeout(() => {
  navigate("/new-post");
}, 200);
```

**Reply by jordanthornquest** [reply]
Will do, thank you!
